### ISSUE TITLE
Hanging and java.lang.NoSuchMethodError

### ISSUE THREAD

**Issue by r0mainK** [src-d/engine#381]
**DISCLAIMER:** this might not be an engine problem

## Expected Behavior

Ran the `apollo preprocess` command on a subfolder of PGA (-> siva files), size of 9.2G. Expected to extract the UAST and other data then convert it to parquet in a reasonable amount of time.

## Current Behavior

It has bow been 89h, and the Spark job has been blocked since friday morning om the second and last stage (first stage took close to 16h). None of the tasks have failed, however I got this warning on the hanging executors:

```
18/04/20 07:13:05 WARN ChannelExecutor: Runnable threw exception in ChannelExecutor
java.lang.NoSuchMethodError: io.netty.handler.codec.http.HttpResponseStatus.codeAsText()Lio/netty/util/AsciiString;
	at io.netty.handler.codec.http2.Http2ConnectionHandler.<clinit>(Http2ConnectionHandler.java:72)
	at io.grpc.netty.NettyClientTransport.start(NettyClientTransport.java:190)
	at io.grpc.internal.ForwardingConnectionClientTransport.start(ForwardingConnectionClientTransport.java:44)
	at io.grpc.internal.InternalSubchannel.startNewTransport(InternalSubchannel.java:217)
	at io.grpc.internal.InternalSubchannel.obtainActiveTransport(InternalSubchannel.java:190)
	at io.grpc.internal.ManagedChannelImpl$SubchannelImplImpl.obtainActiveTransport(ManagedChannelImpl.java:827)
	at io.grpc.internal.GrpcUtil.getTransportFromPickResult(GrpcUtil.java:592)
	at io.grpc.internal.DelayedClientTransport.reprocess(DelayedClientTransport.java:295)
	at io.grpc.internal.ManagedChannelImpl$LbHelperImpl$5.run(ManagedChannelImpl.java:733)
	at io.grpc.internal.ChannelExecutor.drain(ChannelExecutor.java:87)
	at io.grpc.internal.ManagedChannelImpl$LbHelperImpl.runSerialized(ManagedChannelImpl.java:724)
	at io.grpc.internal.ManagedChannelImpl$NameResolverListenerImpl.onAddresses(ManagedChannelImpl.java:773)
	at io.grpc.internal.DnsNameResolver$1.run(DnsNameResolver.java:191)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

As well as the usual:

```
18/04/19 15:34:45 WARN RefWithCommitIterator: missing object for Repository[/spark-temp-data/spark-0e5c8978-10cc-4ef1-b11f-e9fd5afa3bb2/processing-repositories/9A394B8D4D28D4DB00DD680C3485337E/0eeee0bb9d44ddb541b506da1a0bc9f4b4e014b2.siva]; urls https://github.com/c-smile/sciter-sdk
org.eclipse.jgit.errors.MissingObjectException: Missing commit 6e73fdf417a5eea472c1f7c59380e9c85b97b3ba
	at org.eclipse.jgit.internal.storage.file.WindowCursor.open(WindowCursor.java:164)
	at org.eclipse.jgit.revwalk.RevWalk.getCachedBytes(RevWalk.java:903)
	at org.eclipse.jgit.revwalk.RevCommit.parseHeaders(RevCommit.java:155)
	at org.eclipse.jgit.revwalk.RevWalk.markStart(RevWalk.java:293)
	at org.eclipse.jgit.api.LogCommand.add(LogCommand.java:339)
	at org.eclipse.jgit.api.LogCommand.add(LogCommand.java:198)
	at tech.sourced.engine.iterator.RefWithCommitIterator.liftedTree1$1(CommitIterator.scala:155)
	at tech.sourced.engine.iterator.RefWithCommitIterator.hasNext(CommitIterator.scala:153)
	at tech.sourced.engine.iterator.ChainableIterator.hasNext(ChainableIterator.scala:91)
	at scala.collection.Iterator$class.isEmpty(Iterator.scala:330)
	at tech.sourced.engine.iterator.ChainableIterator.isEmpty(ChainableIterator.scala:17)
	at tech.sourced.engine.iterator.ChainableIterator.hasNext(ChainableIterator.scala:80)
	at scala.collection.Iterator$class.isEmpty(Iterator.scala:330)
	at tech.sourced.engine.iterator.ChainableIterator.isEmpty(ChainableIterator.scala:17)
	at tech.sourced.engine.iterator.ChainableIterator.hasNext(ChainableIterator.scala:80)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
	at tech.sourced.engine.iterator.CleanupIterator.hasNext(CleanupIterator.scala:23)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:438)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIterator.sort_addToSorter$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIterator.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anonfun$8$$anon$1.hasNext(WholeStageCodegenExec.scala:395)
	at org.apache.spark.sql.execution.aggregate.SortAggregateExec$$anonfun$doExecute$1$$anonfun$3.apply(SortAggregateExec.scala:80)
	at org.apache.spark.sql.execution.aggregate.SortAggregateExec$$anonfun$doExecute$1$$anonfun$3.apply(SortAggregateExec.scala:77)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$25.apply(RDD.scala:827)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$25.apply(RDD.scala:827)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:335)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

(got a good number of them, but usually it doesn't impact execution).

## Possible Solution

No idea.

## Steps to Reproduce (for bugs)

1. Download data from hdfs://hdfs-namenode/siva/latest/0e (on local-gpu)
2. Try to run a job using or apollo.

## Context

Was preprocessing data for analysis with `apollo`, lost 4 days bc of this job that hung for no apparent reason.

## Your Environment (for bugs)

Pod on local-gpu built using `r0maink/apollo-gpu` image (on Dockerhub), built following the instructions in the guide I PRed in company: https://github.com/src-d/company/pull/122

This is about the 14th repo I tried to preprocess, never had this problem on the others, also never saw hanging never occur at this stage, usually the first stage always hang a bit but not the second.

Im killing the job now because no time to waste more resources, but yeah.

**Comment by erizocosmico** [comment]
Spark and PySpark versions?

**Comment by r0mainK** [comment]
Spark is 2.2.0, Python is 3.4.2, Scala is 2.11.8, engine is 0.5.7

**Comment by r0mainK** [comment]
Seems to be solved with engine `0.6.1`, closing
